sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -484,6 +484,35 @@ case class GpuInitCap(child: Expression) extends GpuUnaryExpression with Implici
     input.getBase.toTitle
 }
 
+case class GpuStringRepeat(strs: Expression, repeatTimes: Expression)
+    extends GpuBinaryExpression with ImplicitCastInputTypes {
+  override def left: Expression = strs
+
+  override def right: Expression = repeatTimes
+
+  override def dataType: DataType = strs.dataType
+
+  override def inputTypes: Seq[AbstractDataType] =
+    Seq(TypeCollection(StringType, BinaryType), IntegerType)
+
+  def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector =
+    throw new UnsupportedOperationException(s"Cannot columnar evaluate expression: $this")
+
+  def doColumnar(lhs: GpuScalar, rhs: GpuColumnVector): ColumnVector =
+    throw new UnsupportedOperationException(s"Cannot columnar evaluate expression: $this")
+
+  def doColumnar(strs: GpuColumnVector, repeatTimes: GpuScalar): ColumnVector = {
+    val repeatTimesVal = repeatTimes.getValue.asInstanceOf[Int]
+    strs.getBase.repeatStrings(repeatTimesVal)
+  }
+
+  def doColumnar(numRows: Int, str: GpuScalar, repeatTimes: GpuScalar): ColumnVector = {
```

Comment:
I don't know how to call the overloaded version of `repeat_strings(string_scalar, int) => string_scalar`. Here I can only call the version that returns a strings column.

Reply:
Looking at #8491 which added the JNI bindings for `repeat_strings`, it appears to be exposed under `Scalar`.  Therefore I think we could avoid exploding the input into a full column by doing something like this:
```scala
    withResource(str.getBase.repeatString(repeatTimes.getValue.asInstanceOf[Int])) { repeatScalar =>
      GpuColumnVector.from(repeatScalar, numRows, StringType)
    }
```